woudc_data_registry/models.py

```diff
@@ -1163,12 +1250,113 @@ def sync(ctx):
+                station_name = station.station_name_id
+
+                """country info"""
+                country_id = station.country_id
+                country_name_en = country_dict[country_id].name_en
+                country_name_fr = country_dict[country_id].name_fr
+
+                """instrument info"""
+                instrument_id = instrument_dict[key].instrument_id
+                instrument_name = instrument_dict[key].name
+                instrument_model = instrument_dict[key].model
+
+                start_date = instrument_dict[key].start_date
+                end_date = instrument_dict[key].end_date
+                dataset_id = instrument_dict[key].dataset_id
+
+                x = instrument_dict[key].x
+                y = instrument_dict[key].y
+                z = instrument_dict[key].z
+
+                """ get the contributor id first from
+                the deployment at the specific station"""
+                contributor_id = \
+                    deployment_dict[station.station_id].contributor_id
+
+                """ now access the project from contributor"""
+                project_id = contributor_dict[contributor_id].project_id
+
+                """form the contribution id by combining the
+                strings present in Contributions dependencies"""
+                contribution_id = project_id + ":" + dataset_id + ":" \
+                    + station_id + ":" + instrument_name
+
+                """check if the contribution id is in the index already"""
+                if contribution_id in contribution_dict.keys():
+
+                    """if it is then update the start
+                    and end date for that contribution id"""
+                    for index in contribution_models:
+                        if contribution_models[index].contribution_id \
+                                == contribution_id:
+                            contribution_models[index].start_date = start_date
+                            contribution_models[index].end_date = end_date
+                else:
+
+                    """otherwise create a new contribution
+                    object with this id and add to the list"""
+                    C = Contribution()
+                    C.contribution_id = contribution_id
+                    C.project_id = project_id
+                    C.dataset_id = dataset_id
+                    C.station_id = station_id
+                    C.station_name = station_name
+                    C.country_id = country_id
+                    C.country_name_en = country_name_en
+                    C.country_name_fr = country_name_fr
+                    C.instrument_id = instrument_id
+                    C.instrument_name = instrument_name
+                    C.instrument_model = instrument_model
+                    C.start_date = start_date
+                    C.end_date = end_date
+                    C.x = x
+                    C.y = y
+                    C.z = z
```

Comment:
Part of the reason we avoid default constructors in Python... lot of lines here.
Also variables should all start with lowercase letters and be more than one letter.